src/sd.c

```diff
@@ -162,7 +162,7 @@ bool sd_write_bin(
     const uint16_t length,
     bool replace)
 {
-    if (!length || length > SD_MAX_FILE_SIZE) {
+    if (!strlens(fn) || data == NULL || !length || length > SD_MAX_FILE_SIZE) {
```

Comment:
Seems wrong to me to return false if length is zero. But that is the current behavior, so we can keep it.

Reply:
Done, PTAL